### PR TITLE
[RV64_DYNAREC] Added C1 /7 SAR opcode

### PR DESCRIPTION
**Pull request by xctan**

### CODE REVIEW

**Comment by ksco** [comment]
I'm not sure, but is there any chance we can simply clear out some flags? e.g. `c >= rex.w ? 64 : 32`?

**Comment by ksco** [comment]
Oh it's a shift right arithmetic, then we need to peek at MSB also, but it might result in fewer RISC-V instructions.

**Comment by xctan** [comment]
Still work in progress, currently 32-bit version is not implemented, and some flags can be optimized.

File: src/dynarec/rv64/dynarec_rv64_emit_shift.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    if(!c) {
+        IFX(X_PEND) {
+            SDxw(s1, xEmu, offsetof(x64emu_t, res));
+        }
+        return;
+    }
+    IFX(X_CF) {
+        SRAI(s3, s1, c-1);
+        ANDI(s3, s3, 1); // LSB
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_emit_shift.c:47]: maybe add a special case when c==1, no neeed for srai, just an andi is enough

**Comment by ptitSeb** [comment]
There is a branch conflict now, sorry. Can you fix it @xctan ?

**Comment by xctan** [comment]
Sure.

File: src/dynarec/rv64/dynarec_rv64_emit_shift.c (hunk 1 above)
Comment (xctan) [reply src/dynarec/rv64/dynarec_rv64_emit_shift.c]: Fixed.

**Review by ksco — APPROVED** [review]

**Pull request merged**